x-pack/plugins/observability/public/embeddable/slo/overview/slo_overview_grid.tsx

```diff
@@ -0,0 +1,185 @@
+    trend: historicalSliData?.map((d) => ({
+      x: d.key as number,
+      y: d.value as number,
+    })),
+    extra: (
+      <FormattedMessage
+        id="xpack.observability.sLOGridItem.targetFlexItemLabel"
+        defaultMessage="Target {target}"
+        values={{
+          target: sloTarget,
+        }}
+      />
+    ),
+    icon: () => <EuiIcon type="visGauge" size="l" />,
+    color: cardColor,
+  };
+};
+
+export function SloCardChartList({ sloId }: { sloId: string }) {
+  const {
+    application: { navigateToUrl },
+    http: { basePath },
+    uiSettings,
+  } = useKibana().services;
+
+  const kqlQuery = `slo.id:"${sloId}"`;
+
+  const { data: sloList, isLoading } = useFetchSloList({
+    kqlQuery,
+  });
+
+  const { data: activeAlertsBySlo } = useFetchActiveAlerts({
+    sloIdsAndInstanceIds: [[sloId, ALL_VALUE]],
+  });
+
+  const { data: rulesBySlo } = useFetchRulesForSlo({
+    sloIds: [sloId],
+  });
+
+  const { data: historicalSummaries = [] } = useFetchHistoricalSummary({
+    list: [
+      {
+        sloId,
+        instanceId: ALL_VALUE,
+      },
+    ],
+  });
+
+  const containerRef = React.useRef<HTMLDivElement>(null);
+
+  const { colors } = useSloCardColor();
+  const chartsData: MetricDatum[][] = [[]];
+  sloList?.results.forEach((slo) => {
+    const subTitle = getSubTitle(slo);
+    const cardColor = colors[slo.summary.status ?? 'NO_DATA'];
+    const { sliValue, sloTarget } = getSloFormattedSummary(slo, uiSettings, basePath);
+
+    const historicalSummary =
+      historicalSummaries.find(
+        (hist) => hist.sloId === slo.id && hist.instanceId === (slo.instanceId ?? ALL_VALUE)
+      )?.data ?? [];
+
+    const lastArray = chartsData[chartsData.length - 1];
+    if (lastArray.length >= 4) {
+      // If the last array has reached its maximum length, create a new array
+      chartsData.push([]);
+    }
+
+    const data = getSloChartData({
+      slo,
+      subTitle,
+      cardColor,
+      sliValue,
+      sloTarget,
+      historicalSummary,
+    });
+    chartsData[chartsData.length - 1].push(data);
+  });
+
+  if (isLoading) {
+    return (
+      <EuiFlexGroup alignItems="center" justifyContent="center" style={{ height: '100%' }}>
+        <EuiFlexItem grow={false}>
+          <EuiLoadingSpinner size="xl" />
+        </EuiFlexItem>
+      </EuiFlexGroup>
+    );
+  }
+
+  return (
+    <div ref={containerRef} style={{ width: '100%' }}>
+      <Chart>
+        <Settings
+          baseTheme={DARK_THEME}
+          onElementClick={([d]) => {
+            if (isMetricElementEvent(d)) {
+              const { columnIndex, rowIndex } = d;
+              const slo = sloList?.results[rowIndex * 4 + columnIndex];
+              const { sloDetailsUrl } = getSloFormattedSummary(slo!, uiSettings, basePath);
+              navigateToUrl(sloDetailsUrl);
+            }
+          }}
+          locale={i18n.getLocale()}
+        />
+        <Metric id={`slo-id-instances`} data={chartsData} />
```

Comment:
Let's make the id unique

Reply:
it's already unique, since it's not a loop.

Reply:
ok yep you are right!